SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -2330,6 +2330,49 @@ public void UpdatePersonSetsCorrectValuesForPersonEntity()
             person.ReviewDate.Should().Be(request.ReviewDate);
         }
 
+        [Test]
+        public void UpdatePersonsAddressUpdatesTheMostRecentAddressToDisplay()
+        {
+            var first = DateTime.Today;
+            var second = DateTime.Today.AddDays(-1);
+            var last = DateTime.Today.AddDays(-2);
+
+            var person = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            person.Addresses = null;
+            var firstAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: first, isDisplayAddress: "Y");
+            var secondAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: second, isDisplayAddress: "Y");
+            var lastAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: last, isDisplayAddress: "Y");
+
+            DatabaseContext.Persons.Add(person);
+            DatabaseContext.Addresses.Add(firstAddress);
+            DatabaseContext.Addresses.Add(secondAddress);
+            DatabaseContext.Addresses.Add(lastAddress);
+            DatabaseContext.SaveChanges();
+
+            var personBeforeUpdate = _classUnderTest.GetPersonDetailsById(person.Id);
+
+            personBeforeUpdate.Addresses.Count.Should().Equals(3);
+            personBeforeUpdate.Addresses.First().StartDate.Should().Equals(first);
+            personBeforeUpdate.Addresses.First().IsDisplayAddress.Should().Equals("Y");
+            personBeforeUpdate.Addresses.First().PostCode.Should().Equals(firstAddress.PostCode);
+
+            var request = GetValidUpdatePersonRequest(person.Id);
+
+            _classUnderTest.UpdatePerson(request);
+
+            var updatedPerson = _classUnderTest.GetPersonDetailsById(person.Id);
+
+            updatedPerson.Addresses.Count.Should().Equals(4);
+            updatedPerson.Addresses.First().StartDate.Should().Equals(first);
+            updatedPerson.Addresses.First().PostCode.Should().Equals(firstAddress.PostCode);
+            updatedPerson.Addresses.First().EndDate.Should().NotBeNull();
+            updatedPerson.Addresses.First().IsDisplayAddress.Should().Equals("N");
+
+            updatedPerson.Addresses.Last().IsDisplayAddress.Should().Equals("Y");
```

Comment:
would it be safer not to rely on the order and identify the new entry by its id? That would ensure the newly added address really is the one that's been set to be the display address? Also might be better to use the db context here as well when fetching the records.